pkg/store/store.go

```diff
@@ -11,14 +11,19 @@ type Value struct {
 	SizeBytes int64
 }
 
-var ErrNotFound = errors.New("not found")
+var (
+	ErrNotFound      = errors.New("not found")
+	ErrQuotaExceeded = errors.New("quota exceeded")
+)
 
 type Store interface {
 	// Get returns the value associated with key.
 	Get(ctx context.Context, key string) (Value, error)
-	// Set associates value with key.
+	// Set associates value with key and returns ErrQuotaExceeded if
+	// that key exceeds quota.
 	Set(ctx context.Context, key string, value Value) error
 	// AddSizeBytes adds to the SizeBytes field of the Value associated
-	// with key.  It creates a new blank Value if needed.
+	// with key and returns ErrQuotaExceeded if that key exceeds quota.
+	// It creates a new blank Value if needed.
 	AddSizeBytes(ctx context.Context, key string, numBytes int64) error
```

Comment:
I don't think we should check for quota and return error on these operations.
I see two flows: (1) Updating the db with resources consumed (2) checking the db upon request.

I think that when separating the 2, we can have a single query to fetch all offending users and be done with it. Doing it thru the actual consumption update is mixing the 2 operations which are triggered differently.

Reply:
There are multiple issues to address in reporting over-quota conditions:

* If we poll by `SELECT`ing every period of time, then the polling period affects the freshness of the results.  I.e. one parameter (polling period) on one service (management service) affects both results freshness on that service and load on another service (Terminus DB).
* If we notify by returning an "over quota" condition on update, then...
   - updates will typically be very bursty.
   - to avoid missed updates we _anyway_ also need to poll.

The future implementation (not what I'm doing right now) can handle _all of these_ by having the management service perform long polls, but Terminus cutting these polls short when something goes over quota.  This error will support that future.

Since it is also not hard to implement, but also simplifies testing, I prefer to give this functionality.